apps/server/src/modules/group/repo/group.repo.ts

```diff
@@ -50,34 +54,87 @@ export class GroupRepo extends BaseDomainObjectRepo<Group, GroupEntity> {
 		return domainObject;
 	}
 
-	public async findByUserAndGroupTypes(user: UserDO, groupTypes?: GroupTypes[]): Promise<Group[]> {
-		let groupEntityTypes: GroupEntityTypes[] | undefined;
+	public async findByUserAndGroupTypes(
+		user: UserDO,
+		groupTypes?: GroupTypes[],
+		query?: IFindQuery
+	): Promise<Page<Group>> {
+		const scope: GroupScope = new GroupScope().byUserId(user.id);
 		if (groupTypes) {
-			groupEntityTypes = groupTypes.map((type: GroupTypes) => GroupTypesToGroupEntityTypesMapping[type]);
+			const groupEntityTypes = groupTypes.map((type: GroupTypes) => GroupTypesToGroupEntityTypesMapping[type]);
+			scope.byTypes(groupEntityTypes);
 		}
 
-		const scope: Scope<GroupEntity> = new GroupScope().byUserId(user.id).byTypes(groupEntityTypes);
+		const escapedName = query?.nameQuery?.replace(MongoPatterns.REGEX_MONGO_LANGUAGE_PATTERN_WHITELIST, '').trim();
+		if (StringValidator.isNotEmptyString(escapedName, true)) {
+			scope.byNameQuery(escapedName);
+		}
 
-		const entities: GroupEntity[] = await this.em.find(GroupEntity, scope.query);
+		const [entities, total] = await this.em.findAndCount(GroupEntity, scope.query, {
+			offset: query?.pagination?.skip,
+			limit: query?.pagination?.limit,
+			orderBy: { name: QueryOrder.ASC },
+		});
 
 		const domainObjects: Group[] = entities.map((entity) => GroupDomainMapper.mapEntityToDo(entity));
 
-		return domainObjects;
+		const page: Page<Group> = new Page<Group>(domainObjects, total);
+
+		return page;
+	}
+
+	public async findAvailableByUser(user: UserDO, query?: IFindQuery): Promise<Page<Group>> {
+		const pipelineStage: unknown[] = [{ $match: { users: { $elemMatch: { user: new ObjectId(user.id) } } } }];
+		const availableGroups: Page<Group> = await this.findAvailableGroup(
+			pipelineStage,
+			query?.pagination?.skip,
+			query?.pagination?.limit,
+			query?.nameQuery
+		);
+
+		return availableGroups;
 	}
 
-	public async findBySchoolIdAndGroupTypes(schoolId: EntityId, groupTypes?: GroupTypes[]): Promise<Group[]> {
-		let groupEntityTypes: GroupEntityTypes[] | undefined;
+	public async findBySchoolIdAndGroupTypes(
+		school: School,
```

Comment:
you could theoretically make a GroupFilter interface that defines attributes like `{ groupTypes, name, schoolId, userId }` and build the scope from that. Then you would only need one funtion instead of findBySchoolIdAndGroupTypes and findByUserAndGroupTypes.

The same goes for findAvailableGroup, where you could use a builder pattern for the pipeline (maybe in a refactoring ticket).